File: src/microscript/Operators.cs
/**
 * MicroScript — The programming language
 * Copyright (c) 2024-2025 Cyril John Magayaga
 * 
 * It was originally written in C# programming language.
 */
namespace com.magayaga.microscript
{
    public class Operators
    {
        public static int Increment(Environment environment, string varName)
        {
            var value = environment.GetVariable(varName);
            if (value is int intValue)
            {
                intValue++;
                environment.SetVariable(varName, intValue);
                return intValue;
            }

            else
            {
                throw new Exception($"Type error: {varName} is not an Integer.");
            }
        }

        public static int Decrement(Environment environment, string varName)
        {
            var value = environment.GetVariable(varName);
            if (value is int intValue)
            {
                intValue--;
                environment.SetVariable(varName, intValue);
                return intValue;
            }

            else
            {
                throw new Exception($"Type error: {varName} is not an Integer.");
            }
        }
    }
}

File: src/com/magayaga/microscript/Operators.java
/**
 * MicroScript — The programming language
 * Copyright (c) 2025 Cyril John Magayaga
 * 
 * It was originally written in Java programming language.
 */
package com.magayaga.microscript;

public class Operators {
    public static int increment(Environment environment, String varName) {
        Object value = environment.getVariable(varName);
        if (value instanceof Integer) {
            int intValue = (Integer) value;
            intValue++;
            environment.setVariable(varName, intValue);
            return intValue;
        }
        
        else {
            throw new RuntimeException("Type error: " + varName + " is not an Integer.");
        }
    }

    public static int decrement(Environment environment, String varName) {
        Object value = environment.getVariable(varName);
        if (value instanceof Integer) {
            int intValue = (Integer) value;
            intValue--;
            environment.setVariable(varName, intValue);
            return intValue;
        }
        
        else {
            throw new RuntimeException("Type error: " + varName + " is not an Integer.");
        }
    }
}